### PR TITLE
Decorator tests

### LINKED ISSUE #2756: Property decorator compiler assertion
Windows 11 23H2
roblox-ts@next

The current version of ``roblox-ts@next`` (7/24/2024) results in a compiler assertion failure whenever I use property decorators. This is noted by @paradoxuum to be caused by the change in transformMethodDeclaration introduced by #2460.

Sample code:
```ts
export function Replicated(target: object, propertyKey: string | symbol) {
    return Reflect.metadata(REPLICATED_METADATA_KEY, true)(target, propertyKey)
}

export default class Components {
    @Replicated
    public static readonly Time = component<{
        isDaytime: boolean,
        secondsElapsed: number,
        daysPassedSinceStart: number,
        nightsPassedSinceStart: number
    }>("Time", {
        isDaytime: true,
        secondsElapsed: 0,
        daysPassedSinceStart: 0,
        nightsPassedSinceStart: 0
    })
}
```

Assertion stack trace:
```
    at assert (c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\Shared\util\assert.js:7:15)
    at transformMemberDecorators (c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\nodes\class\transformDecorators.js:34:33)
    at transformPropertyDecorators (c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\nodes\class\transformDecorators.js:78:12)
    at transformDecorators (c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\nodes\class\transformDecorators.js:116:58)
    at transformClassLikeDeclaration (c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\nodes\class\transformClassLikeDeclaration.js:252:101)
    at transformClassDeclaration (c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\nodes\statements\transformClassDeclaration.js:6:78)
    at transformStatement (c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\nodes\statements\transformStatement.js:76:16)
    at c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\nodes\transformStatementList.js:27:123
    at c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\classes\TransformState.js:132:60
    at TransformState.capturePrereqs (c:\Users\camre.VIRT-MACHINE\OneDrive\Documents\Shared\Projects\2024\Roblox\cmumme\IKEAv2\node_modules\roblox-ts\out\TSTransformer\classes\TransformState.js:127:9)
```

### PR DESCRIPTION
**Pull request by cmumme**
Adds unit tests for:
- Non-static, computed non-static, and static property decorators
- Non-static, computed non-static, and static method decorators
- Non-static & static parameter decorators
- Constructor decorators
- Class decorators

### CODE REVIEW

**Comment by cmumme** [comment]
Need to add tests for non-static property/method/parameter decorators. Working on that now!

**Review by Dionysusnu — CHANGES_REQUESTED** [review]
The changes in transformDecorators.ts are incorrect, commenting out these asserts means the key can become duplicated. Using `member.name.getText()` is incorrect for computed property keys. (In general, getText() is not useful for the compiler, except when debugging)

The change in transformPropertyDeclaration.ts is incorrect, `name` is not the key, but the item being indexed. We should pass the result of `transformPropertyName` instead.

The tests look okay, though it would be good to test computed keys as well.

**Comment by cmumme** [comment]
Added computed key tests and used proper property name fetching in both files. Still have those asserts commented out though, as I'm getting test compile failures without them from that assert in transformMemberDecorators.

**Review by Dionysusnu — CHANGES_REQUESTED** [review]

File: src/TSTransformer/nodes/class/transformPropertyDeclaration.ts [hunk 1]
```diff
@@ -15,6 +15,14 @@ export function transformPropertyDeclaration(
 		return luau.list.make<luau.Statement>();
 	}
 
+	if (ts.hasDecorators(node)) {
+		const propertyName = transformPropertyName(state, node.name);
+
+		if (luau.isSimple(propertyName)) {
+			state.setClassElementObjectKey(node, propertyName);
+		}
```
[the remark this answers is not in the document]
Comment (Dionysusnu) [reply src/TSTransformer/nodes/class/transformPropertyDeclaration.ts:23]: Any kind of property index, eg `[Symbol.iterator]() {}`. For bonus points, one that generates a prereq would be good as an extra unit test too.

**Comment by cmumme** [comment]
I'll modify this PR to only include tests following the merge of #2787.

**Pull request merged**